### PR TITLE
Enable strict concurrency

### PR DESCRIPTION
**Pull request by czechboy0**
### Motivation:

Catch potential data races at build time.

### Modifications:

- Enabled strict concurrency complete checking.
- Created a thread-safe wrapper `SystemMetricsProviderContainer` around `SystemMetricsProvider` to synchronize access to the current provider.
- Changed `CPUUsageCalculator` to be a class, again the previous implementation used a global variable that held a struct, and called a mutating function on it.

### Result:

Fewer potential data races can sneak in.

### Test Plan

Ran tests locally, did not see any more warnings or errors.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -38,3 +38,9 @@ let package = Package(
         ),
     ]
 )
+
+for target in package.targets {
+    var settings = target.swiftSettings ?? []
+    settings.append(.enableExperimentalFeature("StrictConcurrency=complete"))
+    target.swiftSettings = settings
+}
```
Comment (glbrntt) [inline Package.swift:46]: Should we just inline this in the targets?

File: Sources/SystemMetrics/SystemMetrics.swift [hunk 2]
```diff
@@ -41,8 +68,8 @@ extension MetricsSystem {
     ///     - config: Used to configure `SystemMetrics`.
     public static func bootstrapSystemMetrics(_ config: SystemMetrics.Configuration) {
         self.withWriterLock {
-            precondition(self.systemMetricsProvider == nil, "System metrics already bootstrapped.")
-            self.systemMetricsProvider = SystemMetricsProvider(config: config)
+            precondition(self.systemMetricsProviderContainer.isUninitialized, "System metrics already bootstrapped.")
+            self.systemMetricsProviderContainer.bootstrap(SystemMetricsProvider(config: config))
```
Comment (glbrntt) [inline Sources/SystemMetrics/SystemMetrics.swift:72]: Isn't the container using the same lock as we've already acquired here? I.e. we shouldn't acquire the lock here, it should all be done (along with this check) within the container.

**Review by glbrntt — CHANGES_REQUESTED** [review]

File: Package.swift (hunk 1 above)
Comment (czechboy0) [reply Package.swift:46]: I've been using the same snippet for all packages we're working through. Is it worth diverging here?

File: Sources/SystemMetrics/SystemMetrics.swift (hunk 2 above)
Comment (czechboy0) [reply Sources/SystemMetrics/SystemMetrics.swift:72]: You're absolutely right, thanks for catching this deadlock-to-be

Comment (czechboy0) [reply Sources/SystemMetrics/SystemMetrics.swift]: Updated.

**Review by glbrntt — APPROVED** [review]

**Pull request merged**